extensions/binding/org.eclipse.smarthome.binding.iota/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,164 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bindingId="iota" xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+   <!-- IOTA Binding -->
+   <thing-type id="topic" extensible="text,number,percentage,onoff">
+      <supported-bridge-type-refs>
+          <bridge-type-ref id="bridge"/>
+      </supported-bridge-type-refs>
+      <label>Topic</label>
+      <description>This thing represents a topic</description>
+      <config-description>
+         <parameter name="root" type="text" required="true">
+            <label>Root Address</label>
+            <description>The root address from which item states should be retrieved from</description>
+         </parameter>
+         <parameter name="refresh" type="integer" min="5">
+            <label>Refresh interval</label>
+            <description>Specifies the refresh interval in seconds.</description>
+         </parameter>
+         <parameter name="mode" type="text" required="true">
+            <options>
+                <option value="public">Public</option>
+                <option value="private">Private</option>
+                <option value="restricted">Restricted</option>
+            </options>
+            <label>MAM Mode</label>
+            <description>The mode of the channel on which to fetch data</description>
+        </parameter>
+        <parameter name="key" type="text" required="false">
+            <label>Private Key</label>
+            <description>Leave blank for non-restricted mode, otherwise enter the private key you want to use</description>
+         </parameter>
+      </config-description>
+   </thing-type>
+   <thing-type id="wallet" extensible="balance">
+      <supported-bridge-type-refs>
+          <bridge-type-ref id="bridge"/>
+      </supported-bridge-type-refs>
+      <label>IOTA Wallet Information</label>
+      <description>Provides the balances of IOTA wallets</description>
+      <config-description>
+         <parameter name="refresh" type="integer" min="10">
+            <label>Refresh interval</label>
+            <description>Specifies the refresh interval in seconds.</description>
+            <default>10</default>
+         </parameter>
+      </config-description>
+   </thing-type>
+   <channel-type id="balance">
+      <item-type>Number</item-type>
+      <label>Balance</label>
+      <state readOnly="true" pattern="%.5f Miota"> </state>
+      <config-description>
+         <parameter name="address" type="text" required="true">
```

Comment:
this is something we should discuss (read: explain why the modelling is done this way): Instead of having the address on the channel one could also see individual addresses as separate things (with the `address` config parameter on the thing). Why would I want to have several address-channels on one thing? Or is it more intuitive to have a new thing for each address? In case addresses disappear (through snapshots as you pointed out), it would be nice to only have one thing go offline (and with this having a good indication that the address is not available anymore) instead of a channel suddenly not working. wdyt?

Reply:
For the refresh period: done
Note: I fixed a min value of 30 seconds atm. However, in the io bundle, I made it possible that if a refresh value of 0 is used, then synchronous polling is used and data just returns whenever available. This is something we discussed a while ago. As of now I'm not sure this is a feature worth keeping, since it just keeps a thread busy all the time and it doesn't really add any value to the binding. Plus on embedded system it's not really resource efficient. Why want to have the data right away rather than just 30 secs (at most) after they have been published? @htreu I am thinking we should remove that part. Wdyt ?

For the transformation pattern: same remark as in the previous review: how do you tell which data of the array should go to which channel, then? I am thinking if we want to use the transformation pattern as default, we might want to make it as an option in the channel config itself rather than the thing. This way, and considering the user knows what the data look like, he'll be able to tell what data has to go to what channel. Again, wydt?

Finally, for the wallet thing: because of how IOTA signs transactions, an address should only be used once to get funds. From a user POV, wouldn't it be annoying to have tens of things hanging around that are only usable one-time? Specifically if you want to listen to lots of addresses. Imho we should stick with the existing solution.

Reply:
> how do you tell which data of the array should go to which channel, then?

When speaking about JSON, my impression was that JSONPath is able to address very specific single elements. For XML, XPath should do the trick, too.

Reply:
> to have tens of things hanging around that are only usable one-time

Thats in general a problem with the wallet thing / balance channel. For the topic/MAM channel the root address is used, for the balance we have no such alternative. Is there any way to "work around" this issue and make it more user friendly?